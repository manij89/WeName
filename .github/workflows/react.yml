name: React Test
on: [push]
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Install dependencies for client
        run: npm ci --prefix client
      - name: Run tests for client
        run: npm test --prefix client
 