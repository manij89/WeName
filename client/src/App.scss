@import url('https://fonts.googleapis.com/css2?family=Montserrat+Alternates:wght@100;300;400;600&display=swap');

* {
  font-family: 'Montserrat Alternates', sans-serif;
  box-sizing: border-box;

}

