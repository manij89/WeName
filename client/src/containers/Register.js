import React, { useState } from 'react';
import { useForm } from "react-hook-form";
import ErrorMessage from '../components/ErrorMessage';
import '../styles/register.scss';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { registerUser} from '../redux/actions';


function Register(props) {
  const [form, setForm] = useState({
    firstName: '',
    lastName: '',
    email: '',
    password: ''
  });

  const onChange = e => {
    setForm((prev) => {
      const newState = {...prev};
      newState[e.target.name] = e.target.value;
      return newState;
    })
  };

  const {
    register,
    errors,
    formState: { isSubmitting }
  } = useForm();

  const handleSubmit = (e) => {
    e.preventDefault();
    const userData = {
      firstName: form.firstName,
      lastName: form.lastName,
      email: form.email,
      password: form.password
    };
    props.registerUser(userData);
    props.history.push('/game');
  };

  return (
    <div className='registration'>

      <form
        className='registration-form'
        onSubmit={handleSubmit}
        onChange={onChange}
        >
        <h1>Sign Up</h1>
        <label htmlFor='firstName'>First Name:</label>
        <input
          id='firstName'
          className='registration-input'
          name='firstName'
          required />
        <ErrorMessage error={errors.firstName} />

        <label htmlFor='lastName'>Last Name:</label>
        <input
          id='lastName'
          className='registration-input'
          name="lastName"
          required  
          ref={register({ required: true, minLength: 2 })} />
        <ErrorMessage error={errors.firstName} />

        <label htmlFor='email'>Email</label>
        <input
          id='email'
          className='registration-input'
          name="email"
          type="email"
          required 
          ref={register({ required: true, pattern: /^\S+@\S+$/i })}
        />
        <ErrorMessage error={errors.email || errors.validate} />

        <label htmlFor='password'>Password</label>
        <input className='registration-input'
          id='password'
          name="password"
          type="password"
          required 
          ref={register({ required: true, minLength: 8 })}
        />
        <ErrorMessage error={errors.email} />
          <input  data-testid='submit' className='registration-button' disabled={isSubmitting} type="submit" />
        <h5>Already Registered? <Link to='/login'>Login</Link> here</h5>
      </form>

    </div>
  )
}

const mapStateToProps = (state) => ({
  user: state.user
  
})

const mapDispatchToProps = (dispatch) => ({
  registerUser: (userData) => dispatch(registerUser(userData)),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Register);