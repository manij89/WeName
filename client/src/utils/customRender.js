import React from 'react'
import { render } from '@testing-library/react';
import { Provider } from "react-redux";
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import reducer from '../redux/reducers';
import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
const middleware = [thunk];
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;



 const customStore = createStore(
  reducer,
  composeEnhancers(applyMiddleware(...middleware))
);

 const customRender = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store=customStore,
    ...renderOptions
  } = {}) => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <Router history={history}>{children}</Router>
    </Provider>
  )

  return {
    ...render(ui, {
      wrapper: Wrapper,
      ...renderOptions,
    }), 
    history,
    store,
  };
}

export default customRender;
