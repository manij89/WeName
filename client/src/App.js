import './App.scss';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Route, Switch } from 'react-router-dom';
import Home from './containers/Home';
import Register from './containers/Register';
import Login from './containers/Login';
import Game from './containers/Game';
import Profile from './containers/Profile';
import Matches from './components/Matches';
import Deck from './containers/Deck';
import Mylikes from './components/MyLikes';
import Final from './containers/Final';

function App() {
  return (

    <Switch>
      <Route
        exact
        path='/'
        component={Home}
      />
      <Route
        exact
        path='/register'
        component={Register}
      />
      <Route
        exact
        path='/login'
        component={Login}
      />
      <Route
        exact
        path='/mynames'
        component={Mylikes}
      />
      <Route
        exact
        path='/game'
        component={Game}
      />
      <Route
        exact
        path='/final'
        component={Final}
      />
      <Route
        exact
        path='/game/:gender'
        render={routeProps => (
          <>
            <Deck gender={routeProps.match.params.gender} />
          </>
        )}
      />
      <Route
        exact
        path='/profile'
        component={Profile}
      />
      <Route
        exact
        path='/matches'
        component={Matches}
      />
    </Switch>

  );
}

export default App;
