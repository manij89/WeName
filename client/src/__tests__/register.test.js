import React from 'react';
import { cleanup} from '@testing-library/react'
import { createMemoryHistory } from 'history';
import App from '../App';
import Register from '../containers/Register';


import customRender from '../utils/customRender'

afterEach(cleanup)

it('register snapshot', () => {
  const history = createMemoryHistory();
  history.push('/register');
  const { asFragment } = customRender(<App />, { history });
  
  expect(asFragment(<Register />)).toMatchSnapshot()
})