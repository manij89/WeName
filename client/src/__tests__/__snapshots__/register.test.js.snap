// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register snapshot 1`] = `
<DocumentFragment>
  <div
    class="registration"
  >
    <form
      class="registration-form"
    >
      <h1>
        Sign Up
      </h1>
      <label
        for="firstName"
      >
        First Name:
      </label>
      <input
        class="registration-input"
        id="firstName"
        name="firstName"
        required=""
      />
      <label
        for="lastName"
      >
        Last Name:
      </label>
      <input
        class="registration-input"
        id="lastName"
        name="lastName"
        required=""
      />
      <label
        for="email"
      >
        Email
      </label>
      <input
        class="registration-input"
        id="email"
        name="email"
        required=""
        type="email"
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        class="registration-input"
        id="password"
        name="password"
        required=""
        type="password"
      />
      <input
        class="registration-button"
        data-testid="submit"
        type="submit"
      />
      <h5>
        Already Registered? 
        <a
          href="/login"
        >
          Login
        </a>
         here
      </h5>
    </form>
  </div>
</DocumentFragment>
`;
